stl_text/datamodule/doc_classification.py

```diff
@@ -27,17 +49,27 @@ def __init__(self, data_path: str = 'glue_sst2_tiny', vocab_path: Optional[str]
         self.text_transform = None
         self.label_transform = None
         self.datasets = {}
+        self.unk_token = '<unk>'
+        self.tokenizer_type = tokenizer_type
 
     def setup(self, stage):
-        # torchtext's spm_tokenizer and vocab are not pickleable and will fail dataset.map()
-        # tokenizer = spm_tokenizer(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"]))
-        # vocab = build_fairseq_vocab("vocab_tiny.txt")
-
-        # sequential_transforms is not torchscriptable
-        # self.text_transform = sequential_transforms(tokenizer, vocab)
+        if self.tokenizer_type=='sentencepiece':
+            if self.vocab_path:
+                tokenizer = sentencepiece_tokenizer(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"])).to_ivalue()
```

Comment:
The workflow should still work even if you use the pybind version (a.k.a. without `to_ivalue`). Same thing for the vocab blow.